2004/web 4995 kibana document timestamps do not show milliseconds

collin-mcgrath opened the pull request.

collin-mcgrath committed:
Bump aggregate version

collin-mcgrath committed:
When a Time is passed in to a QueryString or MatchCondition, convert it to using 3 decimal places (milliseconds) to make sure it is compatible with the millisecond timestamps that are now being stored in ES

collin-mcgrath committed:
Bump version and update changelog

rjkearns reviewed (approved):
I was also curious if the timestamp creation here should include milliseconds.

The changes look good as is though!

omeedrj commented on CHANGELOG.md:
```diff
@@ -1,6 +1,10 @@
 # Changelog
 All notable changes to this project will be documented in this file
 
+## [1.1.0] - 2020-04-16
```
Same comment applies here about this being a major version bump since the functionality is changing but the interface is not: (https://github.com/Invoca/aggregate/pull/49#discussion_r413181152)

Thoughts?

collin-mcgrath replied:
I actually kind of disagree here. Isn't the idea of this change to support backwards compatible filtering with and without millisecond timestamps?

dtognazzini replied:
I might be missing something, but hasn't the functionality changed? It seems like there could be different behavior based on what version of this gem is used...

omeedrj commented on CHANGELOG.md:
```diff
@@ -1,6 +1,10 @@
+## [1.1.0] - 2020-04-16
+### Added
+- When a `Time` is passed in to a `QueryString` or `MatchCondition`, format it to use 3 decimal places (milliseconds) to guarantee it is compatible with millisecond timestamps stored in ES
+
```
There should also be a `Changed` section indicating that, because of the `aggregate` changes, by default all times will now be stored with milliseconds precision.

collin-mcgrath replied:
I'm not sure I agree. That change is strictly related to `aggregate`. If anything, we could mention that we bumped the version of `aggregate` to make use of the new version, which provides that change. 
Semantics, I know...

omeedrj replied:
> If anything, we could mention that we bumped the version of aggregate to make use of the new version, which provides that change.

That works 👍 As long as the change is pointed out in some way

dtognazzini replied:
I'd say this is a breaking change because the behavior change in `aggregate` manifests here as well.

collin-mcgrath replied:
Ok. I buy that rationale.

omeedrj commented on spec/base_spec.rb:
```diff
@@ -333,7 +333,7 @@ class DummyReplacedModel < DummyElasticSearchModel; end
           time = Time.at(1_544_657_724)
           DummyElasticSearchModel.create!(my_string: "Hello", my_time: time)
-          expect(refresh_and_find_search_hit["_source"]).to eq(@default_fields.merge("my_time" => "2018-12-12T23:35:24Z"))
+          expect(refresh_and_find_search_hit["_source"]).to eq(@default_fields.merge("my_time" => "2018-12-12T23:35:24.000Z"))
```
Similar to [Bobby's comment in the aggregate changes](https://github.com/Invoca/aggregate/pull/49#discussion_r413168333), would we have more peace of mind from using a time with milliseconds? (Same applies to the other test files as well)

collin-mcgrath replied:
Done.

collin-mcgrath committed:
Address CR: Use milliseconds in tests

collin-mcgrath committed:
Bump aggregate dependency to ~> 2.0

collin-mcgrath committed:
Update changelog

dtognazzini commented on spec/query/match_condition_spec.rb:
```diff
@@ -33,14 +33,14 @@
 
       it "converts time to iso8601 and returns range" do
-        time = Time.utc(2018, 12, 27, 20, 10).in_time_zone("Pacific Time (US & Canada)")
+        time = Time.utc(2018, 12, 27, 20, 10, 00, 123).in_time_zone("Pacific Time (US & Canada)")
```
I think this one should be addressed. In Ruby, leading a number with `0` is _saying_ this number is in octal format, it's not a decimal. So... `010` means `8`. Unless there's a specific reason to use octal, we should prefer to use decimal.

collin-mcgrath committed:
Bump a major version since this is a breaking change

collin-mcgrath committed:
Use aggregate version ~> 1.3

collin-mcgrath committed:
- Use the `datetime_formatter` to specify millisecond precision when formatting aggregate datetimes to be stored in ES.

- When a `Time` is passed in to a `QueryString` or `MatchCondition`, use the same format as specified by `datetime_formatter` to guarantee it is compatible the timestamps stored in ES.

collin-mcgrath committed:
Update changelog

dtognazzini commented:
I believe this is superseded by #23. I'm going to unsubscribe from notifications.
